testinfra/modules/file.py

```diff
@@ -171,6 +171,16 @@ def size(self):
     def __repr__(self):
         return "<file %s>" % (self.path,)
 
+    def __eq__(self, other):
+        if isinstance(other, File):
+            return self.path == other.path
+        elif isinstance(other, str):
+            return self.path == other
```

Comment:
I'd use `isinstance(other, six.string_types)` that will also match python 2 unicode string (``isinstance(u'foo', str)`` return ``False`` with python2).

Otherwise your change looks good to me, thanks for implementing this!